### PR TITLE
ci: create gha workflow to automate e2e tests

### PR DESCRIPTION
**Pull request by amisskii**
What issues does this PR fix or reference?

https://github.com/podman-desktop/extension-minikube/issues/139

### CODE REVIEW

**Review by axel7083 — APPROVED** [review]

**Comment by benoitf** [comment]
I want to check the PR before it's being merged, not once it's merged

**Comment by odockal** [comment]
@benoitf That is different issue: https://github.com/podman-desktop/extension-minikube/issues/140.

File: .github/workflows/e2e-main.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        working-directory: ./extension-minikube
+        env:
+          PODMAN_DESKTOP_ARGS: ${{ github.workspace }}/podman-desktop
+        run: pnpm test:e2e
+
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: e2e-tests
+          path: ./**/tests/**/output/
```
Comment (odockal) [inline .github/workflows/e2e-main.yaml:122]: Missing EOF?

File: .github/workflows/e2e-main.yaml [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: e2e-main
+
+on:
+  schedule:
```
Comment (odockal) [inline .github/workflows/e2e-main.yaml:21]: I think it is better to have trigger event "on: push" into main branch.

**Review by odockal — CHANGES_REQUESTED** [review]
Overal is quite ok, just some minor changes, consider adding PR check first so we can see the tests in action.

File: .github/workflows/e2e-main.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        run: pnpm build
+      
+      - name: Ensure getting current HEAD version of the test framework
+        working-directory: ./extension-minikube
+        run: |
+          # workaround for https://github.com/containers/podman-desktop-extension-bootc/issues/712
+          version=$(npm view @podman-desktop/tests-playwright@next version)
+          echo "Version of @podman-desktop/tests-playwright to be used: $version"
+          jq --arg version "$version" '.devDependencies."@podman-desktop/tests-playwright" = $version' package.json > package.json_tmp && mv package.json_tmp package.json
+          pnpm install --lockfile-only
```
Comment (odockal) [inline .github/workflows/e2e-main.yaml:106]: you are calling `pnpm install` twice, here and in the next step.

**Review by odockal — CHANGES_REQUESTED** [review]

**Review by odockal — APPROVED** [review]

**Pull request merged**